### ISSUE TITLE
failure-threshold ignored in configuration file

### ISSUE THREAD

**Issue by huornlmj** [hadolint/hadolint#821]
- [x] This is a bug report
- [ ] This is a feature request
- [X] I searched existing issues before opening this one

### Expected behavior
Only show issues at or above the specified failure-threshold

### Actual behavior
All issues are shown regardless of specified failure-threshold

### Steps to reproduce the behavior
Dockerfile

``` Dockerfile
FROM an.untrusted.registry.local/image:20.04
RUN apt update && \
        apt upgrade -y
RUN pip3 install meson
RUN mkdir directory
RUN wget http://files.site.local/files.zip
```

.hadolint.yaml
``` yaml
trustedRegistries:
        - trusted.domain.local
failure-threshold: warning
```

Command
```console
$ hadolint -V Dockerfile
```

Result
```console
$ hadolint -V Dockerfile
"Configuration file used: REDACTED/.hadolint.yaml"
Configuration:
  no fail: False
  no color: False
  output format: tty
  failure threshold: warning
  override error:
    none
  override warning:
    none
  override info:
    none
  override style:
    none
  override ignore:
    none
  strict labels: False
  label schema:
    none
  disable ignore pragma: False
  allowed registries:
    - - trusted.domain.local
Dockerfile2:1 DL3026 error: Use only an allowed registry in the FROM image
Dockerfile2:2 DL3027 warning: Do not use apt as it is meant to be a end-user tool, use apt-get or apt-cache instead
Dockerfile2:4 DL3013 warning: Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`
Dockerfile2:4 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
Dockerfile2:5 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
Dockerfile2:6 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
Dockerfile2:6 DL3047 info: Avoid use of wget without progress bar. Use `wget --progress=dot:giga <url>`.Or consider using `-q` or `-nv` (shorthands for `--quiet` or `--no-verbose`).
```

**Comment by m-ildefons** [comment]
Hi,
which version of Hadolint are you using? Did this work in a previous version?
Also what is the return code of Hadolint in your example?

This option never changed which rule violations are printed, but it does change the return code. The reason for that is that most cases a return code of 0 indicates success regardless of the output and returncodes != 0 indicate problems.
